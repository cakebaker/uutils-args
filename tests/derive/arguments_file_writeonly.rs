use uutils_args::{Arguments, Options};

#[derive(Arguments)]
#[arguments(file = "/proc/self/clear_refs")] // Oops!
enum Arg {}

struct Settings {}

impl Options<Arg> for Settings {
    fn apply(&mut self, _arg: Arg) -> Result<(), uutils_args::Error> {
        Ok(())
    }
}

fn main() {
    Settings {}.parse(std::env::args_os()).unwrap();
}
