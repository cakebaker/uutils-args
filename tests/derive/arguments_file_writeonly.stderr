error: proc-macro derive panicked
 --> tests/derive/arguments_file_writeonly.rs:3:10
  |
3 | #[derive(Arguments)]
  |          ^^^^^^^^^
  |
  = help: message: cannot open help-string file: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }

error[E0277]: the trait bound `Arg: uutils_args::Arguments` is not satisfied
 --> tests/derive/arguments_file_writeonly.rs:9:6
  |
9 | impl Options<Arg> for Settings {
  |      ^^^^^^^^^^^^ the trait `uutils_args::Arguments` is not implemented for `Arg`
  |
note: required by a bound in `Options`
 --> src/lib.rs
  |
  | pub trait Options<Arg: Arguments>: Sized {
  |                        ^^^^^^^^^ required by this bound in `Options`

error[E0277]: the trait bound `Arg: uutils_args::Arguments` is not satisfied
  --> tests/derive/arguments_file_writeonly.rs:16:17
   |
16 |     Settings {}.parse(std::env::args_os()).unwrap();
   |                 ^^^^^ the trait `uutils_args::Arguments` is not implemented for `Arg`
   |
note: required by a bound in `uutils_args::Options::parse`
  --> src/lib.rs
   |
   | pub trait Options<Arg: Arguments>: Sized {
   |                        ^^^^^^^^^ required by this bound in `Options::parse`
...
   |     fn parse<I>(mut self, args: I) -> Result<(Self, Vec<OsString>), Error>
   |        ----- required by a bound in this associated function
